certbot/error_handler.py

```diff
@@ -93,12 +112,22 @@ def reset_signal_handlers(self):
         self.prev_handlers.clear()
 
     def _signal_handler(self, signum, unused_frame):
-        """Calls registered functions and the previous signal handler.
+        """Stores the recieved signal. If we are executing the code block in
+        the body of the context manager, stop by raising signal exit.
 
         :param int signum: number of current signal
 
         """
-        logger.debug("Singal %s encountered", signum)
-        self.call_registered()
+        self.received_signals.append(signum)
+        if not self.body_executed:
+            raise SignalExit
+
+    def call_signal(self, signum):
+        """Calls the signal given by signum.
```

Comment:
Perhaps we should make this clear that this uses the previous signal handler to handle the signal?

Also, we could perhaps move the looping over the signals to this method? This allows `__exit__` to simply call `self._call_signals()` rather than looping itself.